### ISSUE TITLE
Python bindings are broken

### ISSUE THREAD

**Issue by RevEng-AI** [radareorg/radare2-rlang#37]
Running `configure --prefix=/usr/local && cd python && make` returns the following:

```
Deep recursion on subroutine "main::handle_config" at /usr/bin/pkg-config line 322.
Out of memory!
Deep recursion on subroutine "main::handle_config" at /usr/bin/pkg-config line 322.
Out of memory!
Deep recursion on subroutine "main::handle_config" at /usr/bin/pkg-config line 322.
Out of memory!
```

### Background Info
I'm trying to write a Python plugin for Radare2. I've installed the latest git version of r2 and tried to install r2lang via r2pm, pip, radareorg/radare2-bindings, and this repo - none of which work. I've tried all of the above methods on two different machines running Linux and OpenBSD - same result.

```
r2 -v
radare2 5.7.9 1 @ openbsd-x86-64 git.5.7.9
commit: 7dbd32... build: 2022-10-09__21:39:36
```

**Comment by trufae** [comment]
I think you are looking to the wrong repository. I'll try to clarify all that stuff because it can be a little confusing:

* rlang repository contains "rlang" plugins for radare2. this is, plugins that instantiate the virtual machine or interpreter of X language inside r2, so you can run python scripts and keep global objects alive between multiple executions, but this have nothing to do with the api bindings
* r2pipe allows you to run python scripts as separate processes, not using the api
* the radare2-bindings contains the codegenerators for all languages supported. This is tied to the ABI stability rules, this means that you will need to switch to 5.7.8 or any other patch release to make those bindings work. unless you submit a pr updating the support for the new api that is WIP and will be stable again during the 5.8.x series.

the stack exhaustion error in pkg-config looks to me like a bug in pkg-config, ive never seen that error and i cant reproduce it in local.

The full API bindings of radare2 for python are in radare2-bindings/r2libr. those should build fine, and they are published in pip so you dont need to build them if you use a release version of radare2. for the current (only x.y.9 versions of) radare2 the abi is unstable so you will need to compile it.

Long story short:

* r2pm -ci rlang-python
* r2pm -ci r2api-python

then you may be able to write rlang python plugins using the r2api importing the 'r2libr' python module

**Comment by trufae** [comment]
pip install r2libr will work for stable releases

**Comment by RevEng-AI** [comment]
Using r2libr (Python bindings around C functions/structs) seems like an incredibly tedious way to create a plugin. I might as well write it in C. I can't simply export a RLibStruct because no shared library is created, so I would have to manually hack around and call r_core_plugin_add.

Do you not have a Python API similar to the `r2lang.Plugin` function at the bottom of this  [r2xrefs](https://github.com/radareorg/radare2-extras/blob/master/r2graph/r2xrefs.py) r2 plugin file?

p.s. `r2pm -ci rlang-python` and `r2pm -ci r2api-python` both fail with multiple errors.

```
python.c:322:21: note: (near initialization for ‘r_lang_plugin_python.run_file’)
python/anal.c: In function ‘py_export_anal_enum’:
python/anal.c:107:16: error: ‘R_ANAL_ARCHINFO_INV_OP_SIZE’ undeclared (first use in this function); did you mean ‘R_ANAL_ARCHINFO_MIN_OP_SIZE’?
  107 |         PYENUM(R_ANAL_ARCHINFO_INV_OP_SIZE);
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~
python/anal.c:11:47: note: in definition of macro ‘PYENUM’
   11 |                 PyObject *o = PyLong_FromLong(name); \
      |                                               ^~~~
python/anal.c:107:16: note: each undeclared identifier is reported only once for each function it appears in
  107 |         PYENUM(R_ANAL_ARCHINFO_INV_OP_SIZE);
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~
python/anal.c:11:47: note: in definition of macro ‘PYENUM’
   11 |                 PyObject *o = PyLong_FromLong(name); \
      |                                               ^~~~
python/bin.c: In function ‘Radare_plugin_bin’:
python/bin.c:724:34: warning: assignment to ‘_Bool (*)(RBuffer *)’ {aka ‘_Bool (*)(struct r_buf_t *)’} from incompatible pointer type ‘_Bool (*)(RBinFile *, RBuffer *)’ {aka ‘_Bool (*)(struct r_bin_file_t *, struct r_buf_t *)’} [-Wincompatible-pointer-types]
  724 |                 bp->check_buffer = py_check_buffer;
      |                                  ^
python/io.c: In function ‘Radare_plugin_io’:
python/io.c:204:21: error: ‘RIOPlugin’ {aka ‘struct r_io_plugin_t’} has no member named ‘seek’; did you mean ‘lseek’?
  204 |                 ap->seek = py_io_seek;
      |                     ^~~~
      |                     lseek
python/io.c:210:27: warning: assignment to ‘int (*)(RIODesc *)’ {aka ‘int (*)(struct r_io_desc_t *)’} from incompatible pointer type ‘_Bool (*)(RIODesc *)’
{aka ‘_Bool (*)(struct r_io_desc_t *)’} [-Wincompatible-pointer-types]
  210 |                 ap->close = py_io_close;
      |                           ^
gmake: *** [build.mk:16: lang_python.so] Error 1
```
